src/php/tests/qps/client.php

```diff
@@ -0,0 +1,166 @@
+ * The above complex dance is since threading and async are not idiomatic and we
+ * shouldn't ever be waiting to read a mark
+ *
+ * This test only supports a single channel since threading/async is not idiomatic
+ * This test supports unary or streaming ping-pongs, as well as open-loop
+ * 
+ */
+
+require dirname(__FILE__).'/vendor/autoload.php';
+
+/**
+ * Assertion function that always exits with an error code if the assertion is
+ * falsy.
+ *
+ * @param $value Assertion value. Should be true.
+ * @param $error_message Message to display if the assertion is false
+ */
+function hardAssert($value, $error_message)
+{
+    if (!$value) {
+        echo $error_message."\n";
+        exit(1);
+    }
+}
+
+function hardAssertIfStatusOk($status)
+{
+    if ($status->code !== Grpc\STATUS_OK) {
+        echo "Call did not complete successfully. Status object:\n";
+        var_dump($status);
+        exit(1);
+    }
+}
+
+/* Start the actual client */
+
+function qps_client_main($proxy_address) {
+  echo "Initiating php client\n";
+
+  $proxystubopts = [];
+  $proxystubopts['credentials'] = Grpc\ChannelCredentials::createInsecure();
+  $proxystub = new Grpc\Testing\ProxyClientServiceClient($proxy_address, $proxystubopts);
+  list($config, $status) = $proxystub->GetConfig(new Grpc\Testing\Void())->wait();
+  hardAssertIfStatusOk($status);
+  hardAssert($config->getClientChannels() == 1, "Only 1 channel supported");
+  hardAssert($config->getOutstandingRpcsPerChannel() == 1, "Only 1 outstanding RPC supported");
+
+  echo "Got configuration from proxy, target is " . $config->getServerTargets()[0] . "\n";
+
+  $stubopts = [];
+  if ($config->getSecurityParams()) {
+     if ($config->getSecurityParams()->getUseTestCa()) {
+        $stubopts['credentials'] = Grpc\ChannelCredentials::createSsl(
+            file_get_contents(dirname(__FILE__).'/../data/ca.pem'));
+     } else {
+       $stubopts['credentials'] = Grpc\ChannelCredentials::createSsl(null);
+     }
+     $override = $config->getSecurityParams()->getServerHostOverride();
+     if ($override) {
+     	$stubopts['grpc.ssl_target_name_override'] = $override;
+     	$stubopts['grpc.default_authority'] = $override;
+     }
+  } else {
+    $stubopts['credentials'] = Grpc\ChannelCredentials::createInsecure();
+  }
+  echo "Initiating php benchmarking client\n";
+
+  $stub = new Grpc\Testing\BenchmarkServiceClient($config->getServerTargets()[0], $stubopts);
+  $req = new Grpc\Testing\SimpleRequest();
+
+  echo "Set up php testing request\n";
+
+$req->setResponseType(Grpc\Testing\PayloadType::COMPRESSABLE);
+  $req->setResponseSize($config->getPayloadConfig()->getSimpleParams()->getRespSize());
+  $payload = new Grpc\Testing\Payload();
+  $payload->setType(Grpc\Testing\PayloadType::COMPRESSABLE);
+  $payload->setBody(str_repeat("\0", $config->getPayloadConfig()->getSimpleParams()->getReqSize()));
+  $req->setPayload($payload);
+
+  if (0 && $config->getLoadParams()->getLoad() == "poisson") {
```

Comment:
I think I need to check in with you on this. The PB documentation says that oneofLoad would tell which of the oneof options for variable "load" is being used. The PB generated code indicates that it should be "getLoad," but that ended up not working. What should I be using here? I would love to activate Poisson but couldn't figure out the right incantation to make it work.